### PR TITLE
Languages Endpoint

### PR DESCRIPTION
**Pull request by plyhun**
## Abstract

An endpoint for languages retrieval.

### General

* [x] Added abstract that describes the change
* [x] Added changelog entry to `/CHANGELOG.adoc`
* [x] Ensured that the change is covered by tests
* [x] Ensured that the change is documented in the docs

### On API Changes

* [x] Checked if the changes are breaking or not
* [x] Added GraphQL API if applicable
* [x] Added Elasticsearch mapping if applicable

### CODE REVIEW

File: mdm/common/src/main/java/com/gentics/mesh/core/data/dao/PersistingProjectDao.java [hunk 1]
```diff
@@ -122,6 +128,9 @@ default HibProject create(String name, String hostname, Boolean ssl, String path
 		HibProject project = createPersisted(uuid);
 		project.setName(name);
 
+		// add the default language
```
Comment (plyhun) [inline mdm/common/src/main/java/com/gentics/mesh/core/data/dao/PersistingProjectDao.java:131]: Achtung: do we need to control this?

File: mdm/orientdb-wrapper/src/main/java/com/gentics/mesh/changelog/highlevel/change/AssignLanguagesToProject.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import com.gentics.mesh.core.db.CommonTx;
+import com.gentics.mesh.etc.config.MeshOptions;
+import com.gentics.mesh.graphdb.spi.GraphDatabase;
+import com.gentics.mesh.util.StreamUtil;
+
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+
+@Singleton
+public class AssignLanguagesToProject extends AbstractHighLevelChange {
```
Comment (npomaroli) [inline mdm/orientdb-wrapper/src/main/java/com/gentics/mesh/changelog/highlevel/change/AssignLanguagesToProject.java:31]: This should not be OrientDB specific.

File: mdm/orientdb-wrapper/src/main/java/com/gentics/mesh/changelog/highlevel/change/AssignLanguagesToProject.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	@Override
+	public String getDescription() {
+		return "Checks the languages existing in the project content, and assigns them to the project";
+	}
+
+	@Override
+	public void apply() {
+		CommonTx ctx = CommonTx.get();
+		PersistingLanguageDao languageDao = ctx.languageDao();
+		Map<HibProject, Set<String>> projectLangs = ctx.projectDao().findAll().stream()
```
Comment (npomaroli) [inline mdm/orientdb-wrapper/src/main/java/com/gentics/mesh/changelog/highlevel/change/AssignLanguagesToProject.java:66]: This does not appear to be very efficient: it loads all contents for all nodes for all projects.
Especially when implementing this for Mesh SQL, we need to find a better way to do this.

File: mdm/common/src/main/java/com/gentics/mesh/core/data/dao/PersistingNodeDao.java [hunk 4]
```diff
@@ -2052,6 +2070,8 @@ default void addReferenceUpdates(HibNode updatedNode, BulkActionContext bac) {
 			});
 	}
 
+	Stream<? extends HibNodeFieldContainerEdge> findLanguageEdges(HibProject project, Collection<String> languageTags);
```
Comment (npomaroli) [inline mdm/common/src/main/java/com/gentics/mesh/core/data/dao/PersistingNodeDao.java:2073]: This is only used to determine, whether any of the given languages is currently used in the project.
It is not necessary to return a Stream of all ContainerEdges here, maybe a map of language tag to boolean would be sufficient.
This way, we could find more efficient ways to determine the required information.

**Review by npomaroli — CHANGES_REQUESTED** [review]

File: mdm/orientdb-wrapper/src/main/java/com/gentics/mesh/changelog/highlevel/change/AssignLanguagesToProject.java (hunk 3 above)
Comment (plyhun) [reply mdm/orientdb-wrapper/src/main/java/com/gentics/mesh/changelog/highlevel/change/AssignLanguagesToProject.java:66]: SQL will hold an entirely different implementation.

**Comment by plyhun** [comment]
-> https://github.com/gentics/mesh/pull/1578

**Pull request closed without merging**